sdk/all/src/test/java/io/opentelemetry/sdk/OpenTelemetrySdkTest.java

```diff
@@ -191,4 +195,74 @@ void onlySdkInstancesAllowed() {
         IllegalArgumentException.class,
         () -> OpenTelemetrySdk.builder().setTracerProvider(mock(TracerProvider.class)));
   }
+
+  // This is just a demonstration of maximum that one can do with OpenTelemetry configuration.
+  // Demonstrates how clear or confusing is SDK configuration
+  @Test
+  void fullOpenTelemetrySdkConfigurationDemo() {
+    TraceConfig currentConfig = TraceConfig.getDefault();
+    TraceConfig newConfig =
+        currentConfig.toBuilder()
+            .setSampler(mock(Sampler.class))
+            .setMaxLengthOfAttributeValues(128)
+            .build();
+
+    OpenTelemetrySdk.Builder sdkBuilder =
+        OpenTelemetrySdk.builder()
+            .addSpanProcessor(mock(SpanProcessor.class))
+            .addSpanProcessor(SimpleSpanProcessor.builder(mock(SpanExporter.class)).build())
+            .setIdGenerator(mock(IdGenerator.class))
+            .setPropagators(ContextPropagators.create(mock(TextMapPropagator.class)))
+            .setClock(mock(Clock.class))
+            .setResource(mock(Resource.class));
+
+    TracerSdkProvider tracerSdkProvider =
+        TracerSdkProvider.builder()
+            .setClock(mock(Clock.class))
+            .setIdGenerator(mock(IdGenerator.class))
+            .setResource(mock(Resource.class))
+            .setTraceConfig(newConfig)
+            .build();
+
+    MeterSdkProvider meterSdkProvider =
+        MeterSdkProvider.builder()
+            .setClock(mock(Clock.class))
+            .setResource(mock(Resource.class))
+            .build();
+
+    sdkBuilder.setTracerProvider(tracerSdkProvider);
+    sdkBuilder.setMeterProvider(meterSdkProvider);
+    sdkBuilder.setTraceConfig(newConfig);
+
+    sdkBuilder.build();
+  }
+
+  // This is just a demonstration of the bare minimal required configuration in order to get useful
+  // SDK.
+  // Demonstrates how clear or confusing is SDK configuration
+  @Test
+  void trivialOpenTelemetrySdkConfigurationDemo() {
+    OpenTelemetrySdk.builder()
+        .addSpanProcessor(SimpleSpanProcessor.builder(mock(SpanExporter.class)).build())
+        .setPropagators(ContextPropagators.create(mock(TextMapPropagator.class)))
+        .build();
+  }
+
+  // This is just a demonstration of two small but not trivial configurations.
+  // Demonstrates how clear or confusing is SDK configuration
+  @Test
+  void minimalOpenTelemetrySdkConfigurationDemo() {
+    OpenTelemetrySdk.builder()
+        .addSpanProcessor(SimpleSpanProcessor.builder(mock(SpanExporter.class)).build())
+        .setPropagators(ContextPropagators.create(mock(TextMapPropagator.class)))
+        .setTraceConfig(
+            TraceConfig.getDefault().toBuilder().setSampler(mock(Sampler.class)).build())
+        .build();
```

Comment:
We could turn these all into tests by adding a simple `isNotNull` assert on the results. But, that's only if we feel like we need to have these tests do assertions. Just the fact that they don't throw exceptions is probably enough, TBH.